### ISSUE TITLE
improved_mboxlist_sort: yes is the only sane setting

### ISSUE THREAD

**Issue by elliefm** [cyrusimap/cyrus-imapd#3215]
It's nonsensical to leave this option turned off, but it's off by default, and we keep it like that because changing it requires downing the server to rebuild mailboxes.db.  We're basically catering to people with old deployments who want to upgrade in place with minimal effort, but at the same time we're ensuring that nearly every new deployment ends up stuck with the same bogus default...

Nowadays we don't recommend upgrade-in-place for major releases anyway, instead we tell people to build a new server on the new version with their new configuration and then replicate their data to it.

We've already done big old-world-breaking default changes for unixhierarchysep and altnamespace in 3.0.0, and we plan to do a similar one for virtdomains (#2847), so let's also change this once and for all.

**Comment by elliefm** [comment]
We should probably verify that a server with "improved_mboxlist_sort: no" can successfully replicate to one with "improved_mboxlist_sort: yes" before we go ahead, if that's how we're gonna tell people to upgrade...

**Comment by elliefm** [comment]
Should also verify that the documentation for changing this setting is correct and complete... it lives [here](https://www.cyrusimap.org/imap/developer/thoughts/improved_mboxlist_sort.html), and also somewhat in imapd.conf.5

**Comment by dilyanpalauzov** [comment]
Since `ctl_mboxlist -d -y` provides more output than `ctl_mboxlisd -d`, https://www.cyrusimap.org/imap/developer/thoughts/improved_mboxlist_sort.html shall explicitly state, whether -y shall be passed or not.

**Comment by dilyanpalauzov** [comment]
Note: as far as I remember (see my mail on cyrus-devel from 5th October 2020), exporting the list with ctl_mboxlist -d and importing the list with ctl_mboxlist -u does not work, as suggested in the link above, as the fields `I`, `V` and `F` are missing after the conversion.

**Comment by dilyanpalauzov** [comment]
I have [extended](https://github.com/cyrusimap/cyrus-imapd/pull/3384) ctl_mboxlist to dump/undump all fields.  Following the described procedure: dump and undump on a “dead system” has lead to mboxlist.db doubling its size.  https://www.cyrusimap.org/imap/developer/thoughts/improved_mboxlist_sort.html might need to say, that between the dump and undump mboxlist.db shall be deleted (renamed).

**Comment by hagedose** [comment]
> I have [extended](https://github.com/cyrusimap/cyrus-imapd/pull/3384) ctl_mboxlist to dump/undump all fields. Following the described procedure: dump and undump on a “dead system” has lead to mboxlist.db doubling its size. https://www.cyrusimap.org/imap/developer/thoughts/improved_mboxlist_sort.html might need to say, that between the dump and undump mboxlist.db shall be deleted (renamed).

Confirmed. In our case the undump actually failed with an error message before I realized I had to rename the original mboxlist.db first.

**Comment by elliefm** [comment]
I've updated that doc on the master branch as 41ff335d76, does the new wording make sense?  If so I'll cherry-pick it to the other branches.

**Comment by elliefm** [comment]
there's two things here:

* make "yes" the default instead of "no" -> requires admins to a) know what they have, and b) know whether they need to change it
* make the startup tool whose name i've forgotten right now automatically convert to the improved order on startup -> but this will be a problem for people who upgrade and then need to downgrade for whatever reason

**Comment by dilyanpalauzov** [comment]
If as the startup tool is meant `ctl_mboxlist`, it must be first corrected to export and import all fields from the mailboxes list database.

**Comment by elliefm** [comment]
That's not the tool I meant, but that's a good point.  I can't remember if there is an open issue for that already, I didn't see one on a quick glance just now, but maybe it has a misleading title, or I just didn't look carefully enough.

The startup tool I was referring to is:

```
       START {
           recover       cmd="ctl_cyrusdb -r"
       }
```

**Comment by dilyanpalauzov** [comment]
https://github.com/cyrusimap/cyrus-imapd/pull/3823 is about `ctl_mboxlist` skipping fields in the mailboxes database on import/export.

**Comment by dilyanpalauzov** [comment]
Rectification: https://github.com/cyrusimap/cyrus-imapd/pull/3384 is about `ctl_mboxlist` skipping fields in the mailboxes database on import/export.

**Comment by elliefm** [comment]
This is no longer relevant since 3.6, and can be closed once #4559 lands.

For some reason github won't let me hook it up to close automatically, so if I forget to close it manually, at least this comment explains that I meant to...

**Comment by ksmurchison** [comment]
Done
